k8s-plugin-keel/src/main/kotlin/com/amazon/spinnaker/keel/k8s/verificationEvaluator/K8sJobEvaluator.kt

```diff
@@ -0,0 +1,179 @@
+                }
+                else -> oldState.copy(ConstraintStatus.FAIL).also {
+                    log.warn("verification failed. taskId: $taskId")
+                }
+            }
+        } ?: return oldState.copy(status = ConstraintStatus.FAIL)
+    }
+
+    // This gets invoked if start() was not called for this artifact and environment (null returned from SQL)
+    // metadata (map) is stored in SQL and is passed to evaluate()
+    override fun start(context: ArtifactInEnvironmentContext, verification: Verification): Map<String, Any?> {
+        require(verification is K8sJobVerification) {
+            "verification class must be ${K8sJobVerification::class.simpleName}. received: ${verification.javaClass.simpleName}"
+        }
+        log.info("launching verification K8s job for ${context.deliveryConfig.application}, env: ${context.environmentName}, artifactVersion: ${context.version}")
+        val (job, metadata) = generateOrcaK8sJob(
+            context.deliveryConfig.application,
+            context.environmentName,
+            context.version,
+            verification
+        )
+
+        log.info("this job will be launched in namespace, ${metadata[NAMESPACE]}. jobNamePrefix: ${metadata[GENERATE_NAME]}")
+        return runBlocking {
+            val launcherResponse = taskLauncher.submitJob(
+                user = context.deliveryConfig.serviceAccount,
+                application = context.deliveryConfig.application,
+                notifications = emptySet(),
+                subject = "verify environment, ${context.environmentName}, for ${context.deliveryConfig.application}",
+                description = "verifying ${context.version} in environment ${context.environmentName} with K8s $VERIFICATION_K8S_JOB_V1",
+                correlationId = verification.id,
+                stages = listOf(job),
+                type = VERIFICATION,
+                artifacts = emptyList(),
+                parameters = emptyMap()
+            )
+            metadata += ("taskId" to launcherResponse.id)
+            metadata + ("taskName" to launcherResponse.name)
+        }
+    }
+
+    // need to randomize name otherwise subsequent jobs may fail with the same name in the same namespace
+    // neither orca, clouddriver, nor keel cleans finished jobs.
+    private fun generateOrcaK8sJob(
+        application: String,
+        environment: String,
+        version: String,
+        verification: K8sJobVerification
+    ): Pair<OrcaJob, MutableMap<String, String>> {
+        verification.manifest[API_VERSION] = VERIFICATION_K8S_JOB_API_V1
+        verification.manifest[KIND] = VERIFICATION_K8S_JOB_KIND
+
+        val metadata = if (verification.manifest.containsKey("metadata")) {
+            @Suppress("UNCHECKED_CAST")
+            verification.manifest["metadata"] as MutableMap<String, Any>
+        } else {
+            mutableMapOf()
+        }
+        metadata.remove(NAME)
```

Comment:
rather than removing `NAME`, we could also append it as a suffix to the generated name, maybe? just throwing in some thoughts here. your call.